crates/biome_js_analyze/src/lint/nursery/no_unwanted_polyfillio.rs

```diff
@@ -0,0 +1,273 @@
+use std::sync::LazyLock;
+
+use biome_analyze::{
+    context::RuleContext, declare_lint_rule, Rule, RuleDiagnostic, RuleSource, RuleSourceKind,
+};
+use biome_console::markup;
+use biome_js_syntax::jsx_ext::AnyJsxElement;
+use biome_rowan::{AstNode, TextRange};
+use regex::Regex;
+
+use crate::{
+    nextjs::{is_next_import, NextUtility},
+    services::semantic::Semantic,
+};
+
+declare_lint_rule! {
+    /// Prevent duplicate polyfills from Polyfill.io.
+    ///
+    /// You are using polyfills from Polyfill.io and including polyfills already shipped with Next.js.
+    /// This unnecessarily increases page weight which can affect loading performance.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <script src='https://polyfill.io/v3/polyfill.min.js?features=AbortController,Object.fromEntries'></script>
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// import NextScript from 'next/script';
+    ///
+    /// export function MyApp({ Component, pageProps }) {
+    ///   return <NextScript src='https://polyfill.io/v3/polyfill.min.js?features=Array.prototype.copyWithin' />
+    /// }
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```jsx
+    /// <>
+    ///   <script src='https://polyfill.io/v3/polyfill.min.js?features=AbortController'></script>
+    ///   <script src='https://polyfill.io/v3/polyfill.min.js?features=IntersectionObserver'></script>
+    ///   <Script src='https://polyfill.io/v3/polyfill.min.js?features=IntersectionObserver' />
+    ///   <Script src='https://polyfill-fastly.io/v3/polyfill.min.js?features=IntersectionObserver' />
+    /// </>
+    /// ```
+    ///
+    pub NoUnwantedPolyfillio {
+        version: "next",
+        name: "noUnwantedPolyfillio",
+        language: "jsx",
+        sources: &[RuleSource::EslintNext("no-unwanted-polyfillio")],
+        source_kind: RuleSourceKind::SameLogic,
+        recommended: false,
+    }
```

Comment:
In case you missed the new features, [here the domains](https://github.com/biomejs/biome/blob/next/crates/biome_analyze/CONTRIBUTING.md#rule-domains). 

For this rule, we should assign the `RuleDomain::Next`. And recommend the rule. 

Also, we should [assign a proper severity](https://github.com/biomejs/biome/blob/next/crates/biome_analyze/CONTRIBUTING.md#rule-severity). I leave it to you based on your knowledge.

Plus, we should [add a changeset](https://github.com/biomejs/biome/blob/next/CONTRIBUTING.md#changelog).

Apologies for letting you change things again. Things have been changing a lot :)

Reply:
love the domains feature! I've addressed them at cc081f8 and 057c0ed